src/main/java/com/minecolonies/coremod/util/BackUpHelper.java

```diff
@@ -382,4 +409,164 @@ public static void loadColonyBackup(final int colonyId, final RegistryKey<World>
 
         Log.getLogger().warn("Successfully restored colony:" + colonyId);
     }
+
+    /**
+     * Exports a certain colony and its part of the world and colony data data to a zip
+     *
+     * @param colony colony to export
+     * @return file Path and name
+     */
+    public static String exportColony(final IColony colony)
+    {
+        final MinecraftServer server = ServerLifecycleHooks.getCurrentServer();
+        final File topworldDir = server.func_240776_a_(FolderName.DOT).toFile();
```

Comment:
just side comment, using nio Path would be appreciated as it have things like toAbsolutePath, resolve, normalize and relativize which would remove the replace hacks below

Reply:
yea not worth the effort, would have to do it in the whole file as we use it in other places too